### PR TITLE
added VideoFile object that exposes only part of a video

### PR DESCRIPTION
**Pull request by tonymeng1**
This is an accessor for local video files, required for the splitting video stuff. This object is initialized with a start and end index, but while it is being used, it operates in that original range while pretending it is 0-indexed.

For example, given start=3 and end=6, the object will deal with frames [3,4,5], but API uses [0,1,2] to refer to them. vf.set_index(1) will jump to frame 4 (referenced by 1 in API). In this example, len(vf)=3

It allows fetching the next frame and turning into numpy array. Numpy array dimensions are (frame_num, height, width, rgb)

### CODE REVIEW

**Comment by nicholas-geofroy** [comment]
In this PR could you also remove the numpy array from the VideoItem object? I think it is causing memory issues and this should help fix the problem. Also maybe integrate this with the VideoItem class

**Comment by tonymeng1** [comment]
It looks like that field is being used in the ObjectDetector and the anonomization tests. While the idea was that this class will replace it, I'm not sure I can do it safely. I'm kinda hoping everyone would use this class to replace the parts they individually worked on.

That's also why I requested a review from @Ercadio and @Cvarier - so they can check if as_numpy() will satisfy what they need it for.

**Comment by tonymeng1** [comment]
I don't know if this class can fix the memory issues... I'd assume that the final array size will be same for the full video

File: test/testAnonymizationExecutor.py [hunk 1]
```diff
@@ -42,7 +43,7 @@ def test_face_blurrer_single(self):
 
         # Now we check that the video data has changed
         assert_raises(AssertionError, assert_array_equal, original_data, new_data)
-        
+    """
```
Comment (Cvarier) [inline test/testAnonymizationExecutor.py:46]: 👍

**Review by Cvarier — APPROVED** [review]
lgtm

**Pull request merged**